WordPress/gutenberg
# DataViews: Add sync filter in patterns page

## What?
This PR adds a `sync status` filter in patterns page. 

This PR does a couple of more small things:
1. Adds the ability for elements of fields with type: `enumeration` to provide a helping description
2. Adds an extra check to not a show a filter if no elements are provided. Right now it would render an empty Dropdown

### Design wise
The old patterns page adds an extra header when we apply one of the `sync status` filter. For now I just added that description in the filter menu item, as a helper text. 

## Testing Instructions
1. Enable the admin views experiment and go to patterns page
3. Check that `sync status` filters works as expected
4. Observe that `sync status` field is not visible in sorting and hideable fields menus





## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/16275880/28ede58d-ed95-48f1-9534-818f876617f5
